examples/platforms/simulation/radio.c

```diff
@@ -752,6 +754,7 @@ void radioSendAck(void)
     {
         otEXPECT(otMacFrameGenerateEnhAck(&sReceiveFrame, sReceiveFrame.mInfo.mRxInfo.mAckedWithFramePending, NULL, 0,
                                           &sAckFrame) == OT_ERROR_NONE);
+        otPlatRadioTxAckStarted(aInstance, sAckFrame.mPsdu, sAckFrame.mLength);
```

Comment:
Agree. Discussed with @bukepo, I think we can move forward with the approach for now. I will implement `otPlatRadioSetMacFrameCounter` and `otPlatRadioSetMacKey` for simulation radio so that Enh-ACK will have security and frame counter maintain at platform layer which does not rely on OT APIs and can maintain context dependency there.

Reply:
Ok great. Then we would actually update the Tx frame all in radio layer.